### ISSUE TITLE
allowThen should be enabled in the recommended settings for catch-or-return

### ISSUE THREAD

**Issue by jordanbtucker** [xjamundx/eslint-plugin-promise#168]
### Description

`allowThen` should be enabled in the recommended settings for `catch-or-return`. 

Whether someone decides to use the `.then().catch()` syntax vs the `.then(fn, fn)` syntax is a stylistic choice rather than a problem or anti-pattern. The primary purpose of ESLint is for finding potential issues with code. Recommended configs should not impose stylistic choices upon users.

**Comment by jordanbtucker** [comment]
I'm retracting my statement as the following two pieces of code are not equivalent:

```js
// This will catch the error from the `fulfilled` callback.
Promise.resolve(true)
  .then(result => throw new Error())
  .catch(err => console.error(err))

// This will not catch the error from the `fulfilled` callback.
Promise.resolve(true).then(
  result => throw new Error(),
  err => console.error(err)
)
```
